src/com/google/cose/OkpSigningKey.java

```diff
@@ -99,6 +99,20 @@ public static OkpSigningKey decode(DataItem cborKey) throws CborException, CoseE
     return new OkpSigningKey(cborKey);
   }
 
+  /** Generates a COSE formatted OKP signing key from scratch */
+  public static OkpSigningKey generateKey() throws CborException, CoseException {
+    KeyPair keyPair;
+    try {
+      keyPair = KeyPair.newKeyPair();
+    } catch (GeneralSecurityException e) {
+      throw new CoseException("Error while signing message.", e);
```

Comment:
Nit: I think you forgot to replace the error message here.

Reply:
Replaced, done.